packages/react/src/slider/thumb/SliderThumb.tsx

```diff
@@ -123,6 +125,14 @@ export const SliderThumb = React.forwardRef(function SliderThumb(
     values: sliderValues,
   } = useSliderRootContext();
 
+  if (process.env.NODE_ENV !== 'production') {
+    if (typeof document === 'undefined' && indexProp === undefined && sliderValues.length > 1) {
+      warn(
+        'A `Slider.Thumb` was rendered on the server without an `index` prop, it must be specified for full SSR support.',
+      );
+    }
+  }
```

Comment:
I think it's fine not to have a warning if the demos specify it (and since it's progressive enhancement as mentioned)

Reply:
Maybe, the warning seemed likr a solid one, it didn't seem to have downsides.